apps/test-viewer/src/components/custom-classifications/CustomClassificationsTree.tsx

```diff
@@ -0,0 +1,137 @@
+import { createCachingECClassHierarchyInspector } from "@itwin/presentation-shared";
+import { useClassificationsTreeDefinition } from "@itwin/tree-widget-react";
+
+import type { IModelConnection } from "@itwin/core-frontend";
+
+const PRIMARY_CONNECTION = {
+  iTwinId: "",
+  iModelId: "",
+};
+
+const SECONDARY_CONNECTION = {
+  iTwinId: "",
+  iModelId: "",
+};
+
+export function CustomClassifications(props: { filter?: string }) {
+  const [state, setState] = useState<{ primary: IModelConnection; checkpoint: IModelConnection } | undefined>(undefined);
+
+  useEffect(() => {
+    let disposed = false;
+    async function loadIModels() {
+      const primaryConnection = await CheckpointConnection.openRemote(PRIMARY_CONNECTION.iTwinId, PRIMARY_CONNECTION.iModelId);
+      const checkpointConnection = await CheckpointConnection.openRemote(SECONDARY_CONNECTION.iTwinId, SECONDARY_CONNECTION.iModelId);
+      if (disposed) {
+        return;
+      }
+
+      setState({ primary: primaryConnection, checkpoint: checkpointConnection });
+    }
+
+    void loadIModels();
+    return () => {
+      disposed = true;
+    };
+  }, []);
+
+  if (!state) {
+    return null;
+  }
+
+  const { primary, checkpoint } = state;
+  return <CustomClassificationImpl {...props} primaryIModel={primary} checkpointIModel={checkpoint} />;
+}
+
+function CustomClassificationImpl({
+  primaryIModel,
+  checkpointIModel,
+  filter,
+}: {
+  primaryIModel: IModelConnection;
+  checkpointIModel: IModelConnection;
+  filter?: string;
+}) {
+  const primaryAccess = useMemo(() => {
+    const schemaProvider = createECSchemaProvider(primaryIModel.schemaContext);
+    return {
+      imodelKey: primaryIModel.key,
+      ...createECSchemaProvider(primaryIModel.schemaContext),
+      ...createCachingECClassHierarchyInspector({ schemaProvider, cacheSize: 100 }),
+      ...createLimitingECSqlQueryExecutor(createECSqlQueryExecutor(primaryIModel), 1000),
+    };
+  }, [primaryIModel]);
+
+  const checkpointAccess = useMemo(() => {
+    const schemaProvider = createECSchemaProvider(checkpointIModel.schemaContext);
+    return {
+      imodelKey: checkpointIModel.key,
+      ...createECSchemaProvider(checkpointIModel.schemaContext),
+      ...createCachingECClassHierarchyInspector({ schemaProvider, cacheSize: 100 }),
+      ...createLimitingECSqlQueryExecutor(createECSqlQueryExecutor(checkpointIModel), 1000),
+    };
+  }, [checkpointIModel]);
+
+  const { definition, createFilterPaths } = useClassificationsTreeDefinition({
+    imodelAccesses: useMemo(() => [primaryAccess, checkpointAccess], [primaryAccess, checkpointAccess]),
+    hierarchyConfig: useMemo(
+      () => ({
+        rootClassificationSystemCode: "OpenSite+ Hierarchy",
+      }),
+      [],
+    ),
+  });
+
+  const getFilteredPaths = useCallback(async () => {
+    if (!filter) {
+      return undefined;
+    }
+
+    const [sourcePath, checkpointPaths] = await Promise.all([
```

Comment:
@grigasp I think that building filter paths on text term from all merged iModels should be hidden inside `useClassificationsTreeDefinition`? I expect that `createFilterPaths` will allows to filter by instance keys in the future.

What about showing sub tree + filtering? With current approach that would be on consumer side. I was thinking about hiding this and handling inside similarly how we do with models tree:
```ts
const {} = useClassificationsTreeDefinition({
  imodelAccesses.
  hierarchyConfig,
  getSubTreePaths,
  filter: searchTerm,
});
```